acceptance/tests/environment/environment_scenario-non_existent.rb

```diff
@@ -46,9 +46,9 @@
   },
   :puppet_agent => {
     :exit_code => 1,
-    :matches => [%r{Warning.*404.*Could not find environment '#{env}'},
+    :matches => [%r{(Warning|Error).*(404|400).*Could not find environment '#{env}'},
```

Comment:
here is the output after switching the test back to using the master's environmentpath:
```
huybl681i0j3mx6.delivery.puppetlabs.net (agent) 10:12:05$ puppet agent -t --server tna1l5h77tzypm7.delivery.puppetlabs.net --environment testing --environmentpath='$confdir/environments'
Warning: Unable to fetch my node definition, but the agent run will continue:
Warning: Error 400 on SERVER: Could not find environment 'testing'
Info: Retrieving pluginfacts
Error: /File[/opt/puppetlabs/puppet/cache/facts.d]: Failed to generate additional resources using 'eval_generate': Error 400 on SERVER: Could not find environment 'testing'
Error: /File[/opt/puppetlabs/puppet/cache/facts.d]: Could not evaluate: Could not retrieve file metadata for puppet:///pluginfacts: Error 400 on SERVER: Could not find environment 'testing'
Info: Retrieving plugin
Error: /File[/opt/puppetlabs/puppet/cache/lib]: Failed to generate additional resources using 'eval_generate': Error 400 on SERVER: Could not find environment 'testing'
Error: /File[/opt/puppetlabs/puppet/cache/lib]: Could not evaluate: Could not retrieve file metadata for puppet:///plugins: Error 400 on SERVER: Could not find environment 'testing'
Error: Could not retrieve catalog from remote server: Error 400 on SERVER: Could not find environment 'testing'
Warning: Not using cache on failed catalog
Error: Could not retrieve catalog; skipping run
Error: Could not send report: Error 400 on SERVER: Could not find environment 'testing'
```